cockatrice/src/game/cards/card_database.cpp

```diff
@@ -576,6 +592,63 @@ LoadStatus CardDatabase::loadCardDatabases()
     return loadStatus;
 }
 
+void CardDatabase::refreshPreferredPrintings()
+{
+    for (const CardInfoPtr &card : cards) {
+        card->setPixmapCacheKey(getPreferredPrintingUUIDForCard(card->getName()));
+    }
+}
+
+CardInfoPerSet CardDatabase::getPreferredPrintingForCard(const QString &cardName)
+{
+    CardInfoPtr cardInfo = getCard(cardName);
+    if (!cardInfo) {
+        return CardInfoPerSet(nullptr);
+    }
+
+    CardInfoPerSetMap setMap = cardInfo->getSets();
+    if (setMap.empty()) {
+        return CardInfoPerSet(nullptr);
+    }
+
+    CardSetPtr preferredSet = nullptr;
+    CardInfoPerSet preferredCard;
+    SetPriorityComparator comparator;
+
+    for (auto &set : setMap) {
+        CardSetPtr currentSet = set.getPtr();
+        if (!preferredSet || comparator(currentSet, preferredSet)) {
+            preferredSet = currentSet;
+            preferredCard = set;
```

Comment:
would a break make sense here?

Reply:
I do not think so, since we have to compare through the whole list in case the last entry supersedes the current preferred set.